cvat/apps/consensus/new_intersect_merge.py

```diff
@@ -0,0 +1,1983 @@
+
+    @staticmethod
+    def _merge_anno(a: Iterable[Annotation], b: Iterable[Annotation]) -> List[Annotation]:
+        return merge_annotations_equal(a, b)
+
+    @staticmethod
+    def merge_categories(sources: Iterable[IDataset]) -> CategoriesInfo:
+        return merge_categories(sources)
+
+    @staticmethod
+    def merge_media_types(sources: Iterable[IDataset]) -> Type[MediaElement]:
+        if sources:
+            media_type = sources[0].media_type()
+            for s in sources:
+                if not issubclass(s.media_type(), media_type) or not issubclass(
+                    media_type, s.media_type()
+                ):
+                    # Symmetric comparision is needed in the case of subclasses:
+                    # eg. Image and ByteImage
+                    raise MediaTypeError("Datasets have different media types")
+            return media_type
+
+        return None
+
+
+class _KeypointsMatcher(dm.ops.PointsMatcher):
+    def distance(self, a: dm.Points, b: dm.Points) -> float:
+        a_bbox = self.instance_map[id(a)][1]
+        b_bbox = self.instance_map[id(b)][1]
+        if dm.ops.bbox_iou(a_bbox, b_bbox) <= 0:
+            return 0
+
+        bbox = dm.ops.mean_bbox([a_bbox, b_bbox])
+        return OKS(
+            a,
+            b,
+            sigma=self.sigma,
+            bbox=bbox,
+            visibility_a=[v == dm.Points.Visibility.visible for v in a.visibility],
+            visibility_b=[v == dm.Points.Visibility.visible for v in b.visibility],
+        )
+
+
+def _to_rle(ann: dm.Annotation, *, img_h: int, img_w: int):
+    from pycocotools import mask as mask_utils
+
+    if ann.type == dm.AnnotationType.polygon:
+        return mask_utils.frPyObjects([ann.points], img_h, img_w)
+    elif isinstance(ann, dm.RleMask):
+        return [ann.rle]
+    elif ann.type == dm.AnnotationType.mask:
+        return [mask_utils.encode(ann.image)]
+    else:
+        assert False
+
+
+def segment_iou(
+    a: dm.Annotation,
+    b: dm.Annotation,
+    *,
+    img_h: Union[int, None] = None,
+    img_w: Union[int, None] = None,
+) -> float:
+    """
+    Generic IoU computation with masks and polygons.
+    Returns -1 if no intersection, [0; 1] otherwise
+    """
+    # Comparing to the dm version, this fixes the comparison for segments,
+    # as the images size are required for correct decoding.
+    # Boxes are not included, because they are not needed
+
+    from pycocotools import mask as mask_utils
+
+    is_bbox = AnnotationType.bbox in [a.type, b.type]
+
+    if not is_bbox:
+        assert img_h is not None and img_w is not None
+        a = _to_rle(a, img_h=img_h, img_w=img_w)
+        b = _to_rle(b, img_h=img_h, img_w=img_w)
+    else:
+        a = [list(a.get_bbox())]
+        b = [list(b.get_bbox())]
+
+    # Note that mask_utils.iou expects (dt, gt). Check this if the 3rd param is True
+    return float(mask_utils.iou(b, a, [0]))
+
+
+@attrs
+class IntersectMerge(MergingStrategy):
```

Comment:
Could you inherit the original class and extend it here, instead of copying completely?